ground/src/main/java/com/google/android/ground/persistence/sync/LocalMutationSyncWorker.kt

```diff
@@ -121,30 +118,14 @@ constructor(
     return try {
       mutationRepository.markAsInProgress(mutations)
       remoteDataStore.applyMutations(mutations, user)
-      processPhotoFieldMutations(mutations)
-      mutationRepository.finalizePendingMutations(mutations)
+      mutationRepository.finalizePendingMutationsForMediaUpload(mutations)
```

Comment:
I'd like to refactor the entire MutationsRepository, tbh. I think it would really benefit from a bit more rigidity and consistency. I'm planning on doing that in a separate PR — I can tackle the inlining then.

Reply:
Sgtm!